scripts/build/TestImpactAnalysis/test_impact/base_test_impact.py

```diff
@@ -395,6 +390,22 @@ def _generate_result(self, s3_bucket: str, suite: str, return_code: int, report:
         result["change_list"] = self._change_list
         return result
 
+    def _compile_s3_top_level_dir_name(self, dir_name: str):
+        """
+        Function to build our s3_top_level_dir name. Reads the argument from our dictionary and then appends runtime_type to the end.
+        If s3_top_level_dir name is not provided in args, we will default to "tiaf"+runtime_type.
+
+        @param args: Dictionary containing the arguments passed to this TestImpact object
+
+        @return: Compiled s3_top_level_dir name
+        """
+        if self.runtime_type:
+            if dir_name:
+                dir_name = os.path.join(dir_name, self.runtime_type)
+            else:
+                dir_name = os.path.join("tiaf", self.runtime_type)
```

Comment:
I think we can remove the default `tiaf` and simply raise an exception if `s3_top_level_dir` is not specified when `s3` storage is used.